### PR TITLE
Fixes build script

### PR DESCRIPTION
**Pull request by gbowne1**
This commit: adding .vscode folder and configs.  IT also fixes a flag…problem in the Clang build script ./build.sh by adding '-Wl,-v -static'

### CODE REVIEW

File: src/c/stdlib.c [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #include <stdint.h>
 #include <stdlib.h>
 
-void* malloc(int size) {
+void* malloc(int size_t size)) {
```
Comment (EmptyGroceryBag) [inline src/c/stdlib.c:4]: We haven't implemented `size_t` yet, so you could just put it back to being an `int` or write a `typedef`. Whatever you want to do. Also, there's a trailing `)` at the end of this function signature :)

**Review by EmptyGroceryBag — CHANGES_REQUESTED** [review]

File: src/c/stdlib.c (hunk 1 above)
Comment (gbowne1) [reply src/c/stdlib.c:4]: I am goiñg to try typedef.  

Gonna do some research today too.

Did you have any suggestions?

Comment (gbowne1) [reply src/c/stdlib.c:4]: Ok.  I had some thoughts. 

You're welcome.  I figured add things to make it easier to get started.  

Clang 11 was not that happy with a few things if you look at the output I posted so was aiming at resolving those things especially Malloc size_t

Comment (EmptyGroceryBag) [reply src/c/stdlib.c:4]: > Clang 11 was not that happy with a few things if you look at the output I posted so was aiming at resolving those things especially Malloc size_t

Right, and the problems, as far as I can tell, are only syntax errors. I would change
```c
void* malloc(int size_t size)) {
  return (void*)0x0;
}
```
to
```c
typedef int size_t; // Temporary

void* malloc(size_t size) {
  return (void*)0x0;
}
```
Will that compile on clang 11? I only have access to 14. If you can get that to build, I'll go ahead and merge this.

Also, on my machine, the extra flags you added keeps QEMU from booting from `kernel.bin`. Can you confirm that this happens on your machine too? Also, in the process of finding this out, I found a small bug in the build script related to the iso directory which I will address in a separate commit to this branch.

Comment (EmptyGroceryBag) [reply src/c/stdlib.c:4]: > Also, on my machine, the extra flags you added keeps QEMU from booting from kernel.bin

In particular, it looks like adding the `-static` flag causes QEMU to throw
```
qemu-system-i386: Error loading uncompressed kernel without PVH ELF Note
```

Comment (gbowne1) [reply src/c/stdlib.c:4]: I was going to talk to Strager about that.  He did help a little on a stream a few nights ago towards the end of a stream. You might ask him. I added the flags he suggested 

I have not tested the bin yet.

I still need to upgrade clang but I wanted to see if we could get a clean build between our versions.

I still cant get a response from gcc and tge instructions Ive found for c/c are vague at best.

I do agree with making it a type def.  He said what we have wouldn't work.

I would also suggest adding Makefile, CMakelists and other tooling for people to get going easy.

Comment (gbowne1) [reply src/c/stdlib.c:4]: On the QEMU note

https://forum.osdev.org/viewtopic.php?f=1&t=33638

There were other good links in the search I did on that QEMU error copied to the search.

Comment (gbowne1) [reply src/c/stdlib.c:4]: Oh and I have a stub of a working 386/486 ROM Image dump. Would you like me to include the .bin file?

Comment (gbowne1) [reply src/c/stdlib.c:4]: @EmptyGroceryBag

https://codywu2010.wordpress.com/2014/11/29/about-elf-pie-pic-and-else/

Comment (EmptyGroceryBag) [reply src/c/stdlib.c:4]: > On the QEMU note
> 
> https://forum.osdev.org/viewtopic.php?f=1&t=33638
> 
> There were other good links in the search I did on that QEMU error copied to the search.

Interesting. Maybe we should simply stop using `-kernel` and just pass the ISO file to QEMU.

Comment (gbowne1) [reply src/c/stdlib.c:4]: I don't know much about QEMU itself and I need to read more on it but.. I don't see why not.

File: src/c/stdlib.c [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #include <stdint.h>
 #include <stdlib.h>
 
-void* malloc(int size_t size)) {
+typedef int size_t; // Temporary
+
+void* malloc(size_t size) {
```
Comment (EmptyGroceryBag) [inline src/c/stdlib.c:6]: 👍 Also, I'm looking for a way to keep the `-static` flag when using Clang. It seems to be an issue with using Clang and GNU ld, and the solution looks like using lld with Clang instead.

Comment (gbowne1) [reply src/c/stdlib.c:6]: I made the changes via commits to this that we talked about already.  

Again this was something Strager pointed out.

I need to do some more reading on static.

**Pull request merged**